core/src/main/java/ai/timefold/solver/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -1110,7 +1110,11 @@ public long getMaximumValueRangeSize(Solution_ solution) {
      * @return {@code >= 0}
      */
     public double getProblemScale(Solution_ solution) {
-        long logBase = getMaximumValueRangeSize(solution);
+        long logBase = Math.max(2, getMaximumValueRangeSize(solution));
+        if (logBase == 1 && getListVariableDescriptor() != null && getListVariableDescriptor().allowsUnassignedValues()) {
+            // List variable allowing unassigned values must be increased; otherwise, the scale will be zero.
+            logBase++;
+        }
```

Comment:
This `if` should be removed; it will never trigger (logBase is at least 2).
```suggestion
```

Reply:
Using the minimum of 2 in any situation will result in failure if the value count is zero. Your suggestion makes it simpler, but we still need the if block.

Reply:
Add this line to `ProblemScaleTracker.addBasicProblemScale`:
```
    public void addBasicProblemScale(long count) {
        if (count == 0) {
            // Log(0) = -infinity; also an invalid problem (since there are no
            // valid values for the variable, including null)
            return;
        }
        basicProblemScaleLog += MathUtils.getScaledApproximateLog(MathUtils.LOG_PRECISION, logBase, count);
    }
```